graphs/lca.cpp

```diff
@@ -1,53 +1,65 @@
+/*
+ * Description: Lowest Common Ancestor (LCA) implemented with Binary Lifting.
+ * Supports functionality like kth Parent, distance between pair of nodes in
+ * number of edges, and distance between pair of nodes in sum of edge weight.
+ * Time: O(n * log(n)) for construction and all queries can be answered in
+ * O(log(n))
+ */
+
+
 struct lca {
+	typedef long long ll;
 	int Log;
 	vector<vector<int>> memo;
-	vector<int> depth;
-
-	lca(const vector<vector<int>> &graph, int root) {//0 - based nodes
-		int n = graph.size();
-		Log = 1;
-		while((1<<Log) < n) ++Log;
-		depth.resize(n,0);
-		memo.resize(n,vector<int>(Log));
-		dfs(root,root,graph);
-	}
+	vector<int> depth, dist;
```

Comment:
dist should be a ll. Can you test with a problem where longs e required?

Reply:
Yeah I see, I don't really know one, do you?

Reply:
Hmm, couldn't find one. Can you test instead by writing naive, then running this against naive on random test data that requires longs?

Reply:
And send me the files or results somehow

Reply:
Yeah that'd work, might do this in the morning as I am in the midst of a random Web dev project

Reply:
lol this problem should work https://codeforces.com/contest/1051/problem/F

Reply:
(Don't forget to remove the #define int long long when testing

Reply:
ah that makes sense. I was running with the macro so I guess that's why it worked.

Reply:
https://codeforces.com/contest/1051/submission/124627472 here is the passing test. Can you make sure the code updated, I think it just pulls latest commit from my master until merged?